Update sidebar navigation for version 3.9 of English and Japanese docs

josh-wong opened the pull request:
This navigation is necessary 
> [!CAUTION]
>
> This PR should not be merged until after we're ready to add the `displayed_sidebar: docsEnglish` configuration to the front-matter properties of English docs and `displayed_sidebar: docsJapanese` configuration to the front-matter properties of Japanese docs. By adding this configuration, we can specify the appropriate sidebar language that should be displayed for the respective language when the visitor is on a page.
>
> This method follows Docusaurus's method for sidebars, as described in [Understanding sidebar association](https://docusaurus.io/docs/sidebar/multiple-sidebars#sidebar-association).

## Description

This PR updates the sidebar navigation for ScalarDL 3.9 and 3.8 (currently still supported versions). Since we'll be hard-coding titles (`labels`) in the sidebar navigation in the near future, we should address this now before it becomes an issue.

If we don't specify which sidebar navigation to show for each language, the sidebar navigation for Japanese docs would show in English.

> [!NOTE]
>
> We hard-code the titles (`labels`) in the sidebar navigation so that we can make the titles shorted in the sidebar navigation while keeping longer titles at the top of each doc. Keeping titles within the actual doc provides readers with better context and is better for SEO.

See an example of the sidebar navigation at: 

- https://674ff389d1a359f671379685--scalardl-category-indexes-202412.netlify.app/docs/latest/
  - The sidebar changes exist only for 3.9 docs.

## Related issues and/or PRs

N/A

## Changes made

- Changed the default sidebar name from `docs` to `docsEnglish`.
- Added a sidebar structure for Japanese docs (`docsJapanese`) that matches the English docs.
- Translated the titles (`labels`) in the `docsJapanese` sidebar into Japanese.

## Checklist

> The following is a best-effort checklist. If any items in this checklist are not applicable to this PR or are dependent on other, unmerged PRs, please still mark the checkboxes after you have read and understood each item.

- [x] I have updated the side navigation as necessary.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have updated the documentation to reflect the changes. `N/A`
- [x] Any remaining open issues linked to this PR are documented and up-to-date (Jira, GitHub, etc.).
- [x] My changes generate no new warnings.
- [ ] Any dependent changes in other PRs have been merged and published.

## Additional notes (optional)

N/A

jnmt commented on sidebars.js:
```diff
@@ -161,40 +314,132 @@ const sidebars = {
+                {
+                  type: 'doc',
+                  id: 'helm-charts/getting-started-scalar-helm-charts',
+                  label: 'Scalar Helm Charts by Using a Helm Chart',
```
Seems wordy. How about this?

```suggestion
                  label: 'Use Scalar Helm Charts',
```

jnmt commented on versioned_sidebars/version-3.9-sidebars.json:
```diff
@@ -141,40 +289,132 @@
+                {
+                  "type": "doc",
+                  "id": "helm-charts/getting-started-scalar-helm-charts",
+                  "label": "Scalar Helm Charts by Using a Helm Chart"
```
Ditto.

jnmt commented on sidebars.js:
```diff
@@ -272,8 +601,576 @@ const sidebars = {
+          type: 'doc',
+          id: 'how-to-use-proof',
+          label: 'アセットプルーフを使用する',
+        },
```
I'm not sure, but there aren't the generic contract documents. Is that correct?

jnmt replied:
Thank you!

jnmt reviewed (commented):
Thank you! Almost LGTM. I left a few comments. PTAL!

josh-wong reviewed (commented):
@jnmt Thank you for the review! I've applied your suggestions in 0b2c8fe710e3cde0d91ac45bee5f7350ab8bb5fa and https://github.com/scalar-labs/docs-scalardl/pull/597/commits/f1ff29a3df9753c0088dfc5a37907dd91f9906ad. PTAL again when you have time.

jnmt commented on sidebars.js:
```diff
@@ -272,8 +601,608 @@ const sidebars = {
+        {
+          type: 'doc',
+          id: 'use-generic-contracts',
+          label: '汎用コントラクトおよびファンクションの使用',
```
Probably, this is consistent with others, and a short one might be better.

```suggestion
          label: '汎用コントラクトを使用する',
```

josh-wong replied:
Ah, thank you for mentioning this!

In a PR for the ScalarDB sidebar, [@/feeblefakie mentioned noun form and verb form](https://github.com/scalar-labs/docs-scalardb/pull/762/files/1c1b5404f5b23485105ae6a17a42977ecdc5aac3#r1869009005). In that PR, I changed them to noun form, so I've done the same in the sidebars in this PR as well in https://github.com/scalar-labs/docs-scalardl/pull/597/commits/7e4b2e058e63f387f7222548a350c7d8db97ba74.

jnmt replied:
@josh-wong Thank you! Got it about the noun form thing.

jnmt reviewed (approved):
Thank you for revising it! I left a very minor improvement, but basically LGTM.

josh-wong commented on sidebars.js:
```diff
@@ -63,18 +96,58 @@ const sidebars = {
+        {
+          type: 'doc',
+          id: 'use-generic-contracts',
+          label: 'Use Generic Contracts and Functions',
```
@jnmt As mentioned in [your comment for the Japanese phrasing](https://github.com/scalar-labs/docs-scalardl/pull/597#discussion_r1893676750), should we shorten the English wording as well?


```suggestion
          label: 'Use Generic Contracts',
```

jnmt replied:
@josh-wong Yes, we should do the same.

josh-wong reviewed (commented):
@jnmt I address your comment and made some related changes in https://github.com/scalar-labs/docs-scalardl/pull/597/commits/7e4b2e058e63f387f7222548a350c7d8db97ba74. PTAL!

feeblefakie commented on sidebars.js:
```diff
@@ -92,61 +165,141 @@ const sidebars = {
             },
             {
               type: 'category',
               label: 'Cluster Guides',
```
I don't think `Cluster Guides` and `Container Image Guides` are needed in the sidebar because they are part of deployment and covered by the Deployment guides.

josh-wong replied:
Although I'm OK with not showing these docs in the sidebar in the future, as long as the content is available in the other docs, I do have one concern.

Many other docs link to these docs listed in this sub-category. If we remove these docs from the sidebar, when a visitor clicks a link to go to these pages, then the sidebar navigation will not be displayed on those pages. I think not showing the sidebar navigation would be confusing for users because they would need to go back to the previous page or go to the home page to see the sidebar navigation again.

With that in mind, how about we address this in a different PR?

feeblefakie replied:
I'm fine with working on it in another PR.
I think the second-level categories overall need to be updated to make them similar to the ScalarDB ones.

It's still in the middle of phase 1, right?
I think overall update plans are needed (what's coming next and what should be reviewed in this PR); otherwise, it might be hard to review.

kota2and3kan commented on sidebars.js:
```diff
@@ -272,8 +601,608 @@ const sidebars = {
+        {
+          type: 'doc',
+          id: 'authentication',
+          label: '認証を使用',
```
```suggestion
          label: '認証',
```
I think `認証` is simple and better because the origin English word is `Authentication`.

kota2and3kan commented on sidebars.js:
```diff
@@ -272,8 +601,608 @@ const sidebars = {
+        {
+          type: 'doc',
+          id: 'how-to-handle-errors',
+          label: 'エラーを処理',
```
```suggestion
          label: 'エラーハンドリング',
```
I think `エラーハンドリング` is more general. I think some Japanese engineers use the word `エラーハンドリング` in several blog posts or slide decks.

kota2and3kan commented on sidebars.js:
```diff
@@ -272,8 +601,608 @@ const sidebars = {
+            {
+              type: 'doc',
+              id: 'helm-charts/how-to-deploy-scalar-admin-for-kubernetes',
+              label: 'Kubernetes 用の Scalar Admin をデプロイ',
```
```suggestion
              label: 'Helm Chart を使用して Scalar Admin for Kubernetes をデプロイ',
```
I think `Scalar Admin for Kubernetes` is a product name. So, I think we should not translate `for` here.

kota2and3kan commented on sidebars.js:
```diff
@@ -272,8 +601,608 @@ const sidebars = {
+        {
+          type: 'doc',
+          id: 'installation-with-docker',
+          label: 'ローカル環境に ScalarDL をインストール',
```
```suggestion
          label: 'Docker を使ってローカル環境に ScalarDL をインストール',
```

kota2and3kan commented on sidebars.js:
```diff
@@ -272,8 +601,608 @@ const sidebars = {
+                {
+                  type: 'doc',
+                  id: 'scalar-kubernetes/BackupRDB',
+                  label: 'Kubernetes で RDB をバックアップ',
```
```suggestion
                  label: 'Kubernetes 環境で RDB をバックアップ',
```
I feel `Kubernetes で` has a meaning `by using Kubernetes`. So, I think it would be better to add `環境` here. I think, in Japanese, `Kubernetes 環境で` has meanings like `on Kubernetes` or `in Kubernetes`, but it doesn't have `by using Kubernetes`.

kota2and3kan commented on sidebars.js:
```diff
@@ -272,8 +601,608 @@ const sidebars = {
+                {
+                  type: 'doc',
+                  id: 'scalar-kubernetes/BackupNoSQL',
+                  label: 'Kubernetes で NoSQL データベースをバックアップ',
```
```suggestion
                  label: 'Kubernetes 環境で NoSQL データベースをバックアップ',
```
Ditto.

kota2and3kan commented on sidebars.js:
```diff
@@ -272,8 +601,608 @@ const sidebars = {
+                {
+                  type: 'doc',
+                  id: 'scalar-kubernetes/RestoreDatabase',
+                  label: 'Kubernetes でデータベースを復元',
```
```suggestion
                  label: 'Kubernetes 環境でデータベースを復元',
```
Ditto.

kota2and3kan commented on versioned_sidebars/version-3.9-sidebars.json:
```diff
@@ -10,28 +10,60 @@
+        {
+          "type": "doc",
+          "id": "getting-started-auditor",
+          "label": "Run a Contract Through ScalarDL Ledger and ScalarDL Auditor"
```
```suggestion
          "label": "Run a Contract Through ScalarDL Auditor mode"
```
Same as `sidebars.js`.

josh-wong replied:
Addressed, as mentioned in the https://github.com/scalar-labs/docs-scalardl/pull/597/files#r1897305521 comment thread.

kota2and3kan commented on versioned_sidebars/version-3.9-sidebars.json:
```diff
@@ -141,40 +289,132 @@
+        {
+          "type": "doc",
+          "id": "installation-with-docker",
+          "label": "Install ScalarDL in Your Local Environment"
```
```suggestion
          "label": "Install ScalarDL in Your Local Environment with Docker"
```
Same as `sidebars.js`.

kota2and3kan commented on versioned_sidebars/version-3.9-sidebars.json:
```diff
@@ -197,51 +437,682 @@
+            {
+              "type": "doc",
+              "id": "scalardl-java-client-sdk/README",
+              "label": "ScalarDL Java クライアント SDK"
```
```suggestion
              "label": "ScalarDL Java Client SDK"
```
Same as `sidebars.js`.

kota2and3kan commented on versioned_sidebars/version-3.9-sidebars.json:
```diff
@@ -197,51 +437,682 @@
+        {
+          "type": "doc",
+          "id": "authentication",
+          "label": "認証を使用"
```
```suggestion
          "label": "認証"
```
Same as `sidebars.js`.

kota2and3kan commented on versioned_sidebars/version-3.9-sidebars.json:
```diff
@@ -197,51 +437,682 @@
+        {
+          "type": "doc",
+          "id": "how-to-handle-errors",
+          "label": "エラーを処理"
```
```suggestion
          "label": "エラーハンドリング"
```
Same as `sidebars.js`.

kota2and3kan commented on versioned_sidebars/version-3.9-sidebars.json:
```diff
@@ -197,51 +437,682 @@
+            {
+              "type": "doc",
+              "id": "helm-charts/how-to-deploy-scalar-admin-for-kubernetes",
+              "label": "Kubernetes 用の Scalar Admin をデプロイ"
```
```suggestion
              "label": "Helm Chart を使用して Scalar Admin for Kubernetes をデプロイ"
```
Same as `sidebars.js`.

kota2and3kan commented on versioned_sidebars/version-3.9-sidebars.json:
```diff
@@ -197,51 +437,682 @@
+        {
+          "type": "doc",
+          "id": "installation-with-docker",
+          "label": "ローカル環境に ScalarDL をインストール"
```
```suggestion
          "label": "Docker を使ってローカル環境に ScalarDL をインストール"
```
Same as `sidebars.js`.

kota2and3kan commented on versioned_sidebars/version-3.9-sidebars.json:
```diff
@@ -197,51 +437,682 @@
+                {
+                  "type": "doc",
+                  "id": "scalar-kubernetes/BackupRDB",
+                  "label": "Kubernetes で RDB をバックアップ"
```
```suggestion
                  "label": "Kubernetes 環境で RDB をバックアップ"
```
Same as `sidebars.js`.

kota2and3kan commented on versioned_sidebars/version-3.9-sidebars.json:
```diff
@@ -197,51 +437,682 @@
+                {
+                  "type": "doc",
+                  "id": "scalar-kubernetes/BackupNoSQL",
+                  "label": "Kubernetes で NoSQL データベースをバックアップ"
```
```suggestion
                  "label": "Kubernetes 環境で NoSQL データベースをバックアップ"
```
Same as `sidebars.js`.

kota2and3kan commented on versioned_sidebars/version-3.9-sidebars.json:
```diff
@@ -197,51 +437,682 @@
+                {
+                  "type": "doc",
+                  "id": "scalar-kubernetes/RestoreDatabase",
+                  "label": "Kubernetes でデータベースを復元"
```
```suggestion
                  "label": "Kubernetes 環境でデータベースを復元"
```
Same as `sidebars.js`.

kota2and3kan commented on sidebars.js:
```diff
@@ -272,8 +601,608 @@ const sidebars = {
+        {
+          type: 'doc',
+          id: 'getting-started-auditor',
+          label: 'ScalarDL Ledger と ScalarDL Auditor を通じてコントラクトを実行',
```
```suggestion
          label: 'ScalarDL Auditor モードを通じてコントラクトを実行',
```
If we can use `Auditor mode` as an official word, we can make this label shorter.

josh-wong replied:
Addressed, as mentioned in the https://github.com/scalar-labs/docs-scalardl/pull/597/files#r1897305521 comment thread.

kota2and3kan commented on versioned_sidebars/version-3.9-sidebars.json:
```diff
@@ -197,51 +437,682 @@
+        {
+          "type": "doc",
+          "id": "getting-started-auditor",
+          "label": "ScalarDL Ledger と ScalarDL Auditor を通じてコントラクトを実行"
```
```suggestion
          "label": "ScalarDL Auditor モードを通じてコントラクトを実行"
```
Same as `sidebars.js`.

josh-wong replied:
Addressed, as mentioned in the https://github.com/scalar-labs/docs-scalardl/pull/597/files#r1897305521 comment thread.

kota2and3kan reviewed (commented):
Thank you for the updates! (And sorry for my late review...)
I left several comments and suggestions.
Please take a look when you have time! 🙇‍♂️

feeblefakie commented on sidebars.js:
```diff
@@ -217,52 +462,136 @@ const sidebars = {
+        {
+          type: 'doc',
+          id: 'generic-contracts-reference',
+          label: '汎用コントラクトおよびファンクション',
```
I think we don't treat `Generic contracts` as a proper noun. So, I think it's fine as is.
Maybe, it's better to be just `汎用コントラクト`.

kota2and3kan replied:
@feeblefakie  
Ah, sorry. My comment was a bit ambiguous.

Here is the `docsEnglish` section. Not the `docsJapanese` section.

So, simply we should use English here. I think it was translated by accident.

According to your suggestion, it might be better to use `Generic Contracts` (without `Reference`) here though.

feeblefakie commented on sidebars.js:
```diff
@@ -29,28 +30,60 @@ const sidebars = {
+        {
+          type: 'doc',
+          id: 'getting-started-auditor',
+          label: 'Run a Contract Through ScalarDL Ledger and ScalarDL Auditor',
```
Casually saying `Auditor mode` is fine, but I think it's not a mode.
We can say something like `ScalarDL Ledger and Auditor` or  `ScalarDL with Auditor`.

```suggestion
          label: 'Run a Contract Through ScalarDL Ledger and Auditor',
```

kota2and3kan replied:
@feeblefakie 
I see. I understood. Thank you for your comment!
I feel `ScalarDL with Auditor` is good because it's short and simple, but both are fine to me. :)

feeblefakie reviewed (approved):
For just introducing sidebar.js and stop hard-coding the sidebar menus, it's fine.
In that case, let's not update the content at all.

Anyway, I think all the sidebar menus need to be updated soon.

jnmt reviewed (approved):
LGTM, thank you!

josh-wong commented:
@kota2and3kan I've addressed your comments and others that we discussed in today's short meeting (https://github.com/scalar-labs/docs-scalardl/pull/597/commits/0368d938b5ee74cf70a81247c10f812a21958513). PTAL!
